plugin/formatting.py

```diff
@@ -100,16 +102,70 @@ class LspFormatDocumentCommand(LspTextCommand):
 
     capability = 'documentFormattingProvider'
 
-    def is_enabled(self, event: Optional[dict] = None, point: Optional[int] = None) -> bool:
+    def is_enabled(self, event: Optional[dict] = None, select: bool = False) -> bool:
+        if select:
+            return len(list(self.sessions(self.capability))) > 1
         return super().is_enabled() or bool(self.best_session(LspFormatDocumentRangeCommand.capability))
 
-    def run(self, edit: sublime.Edit, event: Optional[dict] = None) -> None:
-        format_document(self).then(self.on_result)
+    def run(self, edit: sublime.Edit, event: Optional[dict] = None, select: bool = False) -> None:
+        session_names = [session.config.name for session in self.sessions(self.capability)]
+        base_scope = self.view.scope_name(0).split()[0]
+        if select:
+            self.select_formatter(base_scope, session_names)
+        elif len(session_names) > 1:
+            window = self.view.window()
+            if not window:
+                return
+            formatter = None
+            project_data = window.project_data()
+            if isinstance(project_data, dict):
+                formatter = project_data.get('settings', {}).get('LSP', {}).get('formatters', {}).get(base_scope)
+            else:
+                window_manager = windows.lookup(window)
+                if window_manager:
```

Comment:
Can we do this in place of `window = self.view.window()` already? In theory we should always go through WindowManager to verify that LSP is still operational.

Reply:
We still need the window for `window.project_data()` a few lines above. So I don't think it would simplify the code here. And this `if window_manager` is just to satisfy pyright type checking, otherwise it shouldn't really be needed because the command wouldn't run anyway (`is_enabled` would return false) if there is no window_manager / LSP active. So I think it's most useful to defer this `lookup` line into the else branch and only do that function call if necessary. Or did you mean something else?

Reply:
We can get window from `wm.window`.

And yes, you are right that in that particular case it shouldn't be possible for window to be missing but it wouldn't hurt to always assume that it can instead of picking on a case by case basis whether we should be extra safe.

Reply:
Moved to a separate function. I think it should be okay now.